SpatialGDK/Source/SpatialGDK/Private/Utils/SettingUtils.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SettingUtils.h"
+#include "SpatialConstants.h"
+#include "SpatialPersistenceConfig.h"
+#include "SpatialWorldSettings.h"
+
+namespace SpatialGDK
+{
+bool ShouldClassPersist(const ASpatialWorldSettings* SpatialWorldSettings, const UClass* InClass)
+{
+	// If persistence settings don't exist, persist everything by default
+	if (SpatialWorldSettings == nullptr)
+	{
+		return true;
+	}
+
+	const USpatialPersistenceConfig* SpatialPersistenceConfig = SpatialWorldSettings->GetSpatialPersistenceConfig();
+
+	if (SpatialPersistenceConfig == nullptr)
+	{
+		return true;
+	}
+
+	const EPersistenceSelectionMode Mode = SpatialPersistenceConfig->GetPersistenceSelectionMode();
```

Comment:
Any reason this logic isn't in USpatialPersistenceConfig ?

Reply:
I had it like that originally...but in my cleanup pass, I noticed that it seems in general the UDataAsset extension classes avoid adding logic - treating them more as pure data classes.

I can go back to that approach if you prefer it.